packages/loopring_v3/contracts/amm/libamm/AmmBlockReceiver.sol

```diff
@@ -74,11 +81,35 @@ library AmmBlockReceiver
         }
 
         // The closing AMM updates
-        S.processAmmUpdates(ctx, false);
+        S.approveAmmUpdates(ctx, false);
+
+        // Save the pool tokens to burn
+        uint totalSupply = S.totalSupply;
+
+        if (ctx.totalSupply > totalSupply) {
+            S.mint(address(this), ctx.totalSupply - totalSupply);
+        } else {
+            S.poolTokenToBurn = totalSupply - ctx.totalSupply;
+        }
 
         return ctx.numTransactionsConsumed;
     }
 
+    function afterBlockSubmission(
```

Comment:
Maybe easier/more efficient to just modify `totalSupply()` to return `totalSupply.sub(totalBurned)` (with maybe a function to actually burn the tokens and resetting `totalBurned` back to 0, though shouldn't really be needed because uint256 is plenty)?

Reply:
So `afterAllBlocksSubmitted` can now be removed because it's not necessary any more?

Reply:
Ah maybe there was some misunderstanding. I was just trying to avoid having to do another callback afterwards, I was fine with how the caching worked before. But now that only a single transaction can be processed it probably doesn't matter that much any more either way.